From bc57bc9e77829c567155ce2c61e84e91385214fc Mon Sep 17 00:00:00 2001
From: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Date: Wed, 12 Jun 2013 13:13:07 +0300
Subject: [PORT FROM MAIN] telephony: Handle data disabled after voice call ended

BZ: 105930

If user disables data during data suspended state
(2G voice call), data deactivation request is not
processed. So, when the voice call is ended, data
icon is still shown.

Once the voice is ended, check the mobile data
enabled state. If mobile data is enabled, then
cleanup all data connections.

Change-Id: Ib0c6c0b45f5625367e8bab9592507453b3b1ac81
Orig-Change-Id: I289d55073d97aff211d8ea675c2073771779c5fb
Signed-off-by: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../telephony/dataconnection/DcTracker.java        |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/src/java/com/android/internal/telephony/dataconnection/DcTracker.java b/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
index 52f0eb7..1ea2e8e 100644
--- a/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
+++ b/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
@@ -1748,6 +1748,10 @@ public final class DcTracker extends DcTrackerBase {
         if (DBG) log("onVoiceCallEnded");
         mInVoiceCall = false;
         if (isConnected()) {
+            if (!getAnyDataEnabled()) {
+                 onCleanUpAllConnections(Phone.REASON_DATA_DISABLED);
+                return;
+            }
             if (!mPhone.getServiceStateTracker().isConcurrentVoiceAndDataAllowed()) {
                 startNetStatPoll();
                 startDataStallAlarm(DATA_STALL_NOT_SUSPECTED);
-- 
1.7.4.1

